policy/file/source.go

```diff
@@ -94,16 +96,10 @@ func (s *Source) MonitorIDs(ctx context.Context, resultCh chan<- policy.IDMessag
 		case <-s.reloadChan:
 			s.log.Info("file policy source ID monitor received reload signal")
 
-			// We are reloading all files within the directory so wipe our
-			// current mapping data.
-			s.idMapLock.Lock()
-			s.idMap = make(map[pathMD5Sum]policy.PolicyID)
-			s.idMapLock.Unlock()
-
 			s.identifyPolicyIDs(resultCh, errCh)
 
 			// Tell the MonitorPolicy routines to reload their policy.
-			s.policyReloadChan <- true
+			s.policyReloadChan <- struct{}{}
```

Comment:
I am probably missing something, but wouldn't you have to send multiple messages in this channel to trigger a reload in all `MonitorPolicy` routines?

Reply:
Ah yeah great shout, not sure how I missed that one!